tools/c7n_azure/c7n_azure/resources/subscription.py

```diff
@@ -71,6 +73,89 @@ def _get_subscription(self, session_factory, config):
 
 Subscription.filter_registry.register('missing', Missing)
 
+@Subscription.filter_registry.register('diagnostic-settings')
+class SubscriptionDiagnosticSettingFilter(ValueFilter):
+    """Filter by diagnostic settings for this subscription
+
+    Each diagnostic setting for the subscription is made available to the filter. The data format
+    is the result of making the following Azure API call and extracting the "value" property:
+    https://learn.microsoft.com/en-us/rest/api/monitor/subscription-diagnostic-settings/list?tabs=HTTP
+
+    :example:
+
+    Example JSON document showing the data format provided to the filter
+
+    .. code-block:: json
+        {
+          "id": "...",
+          "name": "...",
+          "properties": {
+            "eventHubAuthorizationRuleId": "...",
+            "eventHubName": "...",
+            "logs": [
+              { "category": "Administrative", "enabled": true },
+              { "category": "Security", "enabled": false }
+            ],
+            "marketplacePartnerId": "...",
+            "serviceBusRuleId": "...",
+            "storageAccountId": "...",
+            "workspaceId": "..."
+          },
+          "systemData": {}
+          "type": "..."
+        }
+
+    :example:
+
+    Check if the subscription has Security logs enabled in at least one setting
+
+    .. code-block:: yaml
+
+        policies:
+          - name: subscription-security-logs-enabled
+            resource: azure.subscription
+            filters:
+              - not:
+                - type: diagnostic-settings
+                  key: "properties.logs[?category == 'Security'].enabled[]"
+                  op: contains
+                  value: true
+
+    """
+
+    cache_key = 'c7n:diagnostic-settings'
```

Comment:
not used afaics.. it might be good to have annotation key just for the matched diagnostic settings, on the resource.

Reply:
Yep, I changed tactics part way through and failed to carry it forward. I've updated the code to make use of it now. Thanks!